### PR TITLE
[controller & command] calc fpfn before visualization task

### PR DESCRIPTION
**Pull request by fenrir-z**
- [x] app: pass iou_thr and conf_thr to visualization invoker
- [x] visualization grpc api: add iou_thr and conf_thr to VisualizationRequest
- [x] visualization invoker: call evaluate command before export, pass iou_thr and conf_thr
- [x] evaluate command: add --calc-confusion-matrix
- [x] det_eval: calc fpfn
- [ ] det_eval: generate pred annotations in MirAnnotations with tags: {'confusion_matrix': 'fp'} or {'confusion_matrix': 'fn'} or  {'confusion_matrix': 'tp'} 
- [ ] test cases for app / controller / command / det_eval

### CODE REVIEW

File: ymir/backend/src/ymir_controller/controller/utils/utils.py [hunk 1]
```diff
@@ -67,6 +68,17 @@ def sub_task_id(task_id: str, offset: int) -> str:
     return task_id[0] + str(offset) + task_id[2:]
 
 
+def gen_task_id(user_id: int, project_id: int) -> str:
```
Comment (phoenix-xhuang) [inline ymir/backend/src/ymir_controller/controller/utils/utils.py:71]: 使用https://github.com/IndustryEssentials/ymir/blob/master/ymir/backend/src/common/id_definition/task_id.py#L30

File: ymir/backend/src/ymir_controller/controller/invoker/invoker_task_visualization.py [hunk 2]
```diff
@@ -56,17 +60,32 @@ def subtask_invoke_0(
...
+        repo_root: str,
+        work_dir: str,
+        media_location: str,
+        fmt: str,
+        iou_thr: float,
+        conf_thr: float,
+        dataset_id: str,
+    ) -> str:
+        # evaluate and add fpfn
+        evaluated_dataset_id = utils.gen_task_id(user_id=0, project_id=0)
```
Comment (phoenix-xhuang) [inline ymir/backend/src/ymir_controller/controller/invoker/invoker_task_visualization.py:81]: 看上去应该把task_id传进来，然后在这里拼为dataset_id@task_id

Comment (phoenix-xhuang) [reply ymir/backend/src/ymir_controller/controller/invoker/invoker_task_visualization.py:81]: 在新的pr里面已经更正了
https://github.com/IndustryEssentials/ymir/pull/846

**Comment by phoenix-xhuang** [comment]
这个PR可以合并了，在这个基础上完成的pr846会完成剩余的工作
https://github.com/IndustryEssentials/ymir/pull/846

**Review by phoenix-xhuang — APPROVED** [review]

**Review by IJtLJZ8Rm4Yr — APPROVED** [review]

**Pull request merged**